common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAuthority.java

```diff
@@ -173,4 +174,21 @@ public AzureActiveDirectoryAudience getAudience() {
         return mAudience;
     }
 
+    /**
+     * Checks if current authority belongs to the same cloud as the passed in authority
+     *
+     * @param authorityToCheck authority to check against
+     * @return true if host name matches for authorityURL of both authorities, otherwise false
+     */
+    public boolean isSameCloudAsAuthority(@NonNull final AzureActiveDirectoryAuthority authorityToCheck) throws IOException {
+        if (!AzureActiveDirectory.isInitialized()) {
+            // Cloud discovery is needed in order to make sure that we have a preferred_network_host_name to cloud aliases mappings
+            AzureActiveDirectory.performCloudDiscovery();
+        }
+
+        // getAuthorityURL returns a url with mapped preferred_network_host_name as the host part of the url
+        // So for two authorities from same cloud getAuthorityURL should return the same host name
+        return this.getAuthorityURL().getHost().equals(authorityToCheck.getAuthorityURL().getHost());
```

Comment:
```suggestion
        return getAuthorityURL().getHost().equals(authorityToCheck.getAuthorityURL().getHost());
```

Doesn't appear to be needed